qcodes/utils/validators.py

```diff
@@ -511,20 +514,31 @@ def __init__(self, min_value: numbertypes=-float("inf"),
             self._max_value = max_value
         else:
             raise TypeError('max_value must be a number bigger than min_value')
-        self._shape = shape
+        if not isinstance(shape,
+                          collections.abc.Sequence) and shape is not None:
+            raise ValueError(f"Shape must be a sequence (List, Tuple ...) "
+                             f"got a {type(shape)}")
+        self._shape: shape_tuple_type = None
+        if shape is not None:
+            self._shape = tuple(shape)
+
 
     @property
     def valid_values(self) -> Tuple[np.ndarray]:
         shape = self.shape
         if shape is None:
             return (np.array([self._min_value]),)
         else:
-            val_arr = np.empty(shape)
+            val_arr = np.empty(self.shape_evaluated)
             val_arr.fill(self._min_value)
             return (val_arr,)
 
     @property
-    def shape(self) -> Optional[Tuple[int, ...]]:
+    def shape(self) -> shape_tuple_type:
```

Comment:
will this break anything? I know that tests don't fail, but in general? For example, would it be safer to leave `shape` as is, and add `shape_unevaluated` or `shape_raw`?

Reply:
I don't think it's an issue but I agree that the other way around is safer so lets just do that